### ISSUE TITLE
terminate called after throwing an instance of 'std::bad_alloc'

### ISSUE THREAD

**Issue by joshuagi** [alexdobin/STAR#237]
Hi, 

I am using STAR-2.5.2b/bin/Linux_x86_64_static/STAR to generate genome indices for alignment. I entered the following command:

 ~/STAR-2.5.2b/bin/Linux_x86_64_static/STAR --runThreadN 25 --runMode genomeGenerate --genomeDir ./ --genomeFastaFiles ./Papio_anubis.PapAnu2.0.dna.toplevel.fa --sjdbGTFfile ./Papio_anubis.PapAnu2.0.87.gtf --limitGenomeGenerateRAM 1000000000000

and I got the following error:

Feb 13 19:27:23 ..... started STAR run
Feb 13 19:27:23 ... starting to generate Genome files
Feb 13 19:32:43 ... starting to sort Suffix Array. This may take a long time...
Feb 13 19:35:59 ... sorting Suffix Array chunks and saving them to disk...
terminate called after throwing an instance of 'std::bad_alloc'
terminate called recursively
terminate called recursively
Aborted

I'm pretty new to RNAseq analysis... I would appreciate any help!! :)

Josh

**Comment by alexdobin** [comment]
Hi @joshuagi 

What is the RAM available on your machine? Also please send me the Log.out file of this run - I will try to suggest the parameters to avoid this error.

Cheers
Alex

**Comment by joshuagi** [comment]
Hi Dr. Dobin,

Thanks so much for your prompt reply! I tried running the genome generate option with just one chromosome instead of the full assembly, and it worked. I am now (since about an hour) sure that it is a problem with my RAM. I’m sorry I was about to update my post to indicate that this issue is closed.

I have talked with some of my colleagues and there is another server I can try. In a sense, I am “learning how to learn” and before I post again on your forum I will exhaust all my options and try different things and try to troubleshoot it myself.

Kind regards,

Josh
